### PR TITLE
Server configuration handling improvements

### PR DESCRIPTION
**Pull request by johngian**
* Add dotenv dependency
* Update config setup to allow loading dotenv
* Replace local file-based docker setup with env vars or dotenv
* Add compabitility with legacy JSON config (legacy nubis support)

### CODE REVIEW

File: server/src/config-helper.ts [hunk 1]
```diff
@@ -86,14 +100,18 @@ export function getConfig(): CommonVoiceConfig {
...
+      let config_path = process.env.SERVER_CONFIG_PATH || './config.json';
+      fileConfig = JSON.parse(fs.readFileSync(config_path, 'utf-8'));
+    } catch (err) {
+      console.error(err, 'could not load config.json, using defaults');
+    }
   }
-  loadedConfig = { ...DEFAULTS, ...config };
 
-  return loadedConfig;
+  loadedConfig = {...BASE_CONFIG, ...fileConfig}
```
[the remark this answers is not in the document]
Comment (The-smooth-operator) [reply server/src/config-helper.ts]: In my opinion env vars should always override `fileConfig`

[the remark this answers is not in the document]
Comment (johngian) [reply server/src/config-helper.ts]: This fileConfig is only there for Nubis. So in that case we want to have a fully file-based setup. After the nubis decommision dotenv does all the handling.

* Given an env var --> load a config file
* Env vars have priority

**Comment by phirework** [comment]
Looks good to me, feel free to merge.

**Comment by johngian** [comment]
@The-smooth-operator just a heads up: you can use this build instead of my temp docker image

**Comment by The-smooth-operator** [comment]
thanks @johngian I can confirm it works!!

**Pull request merged**